src/network/netdev/wireguard.c

```diff
@@ -527,6 +528,38 @@ int config_parse_wireguard_private_key(const char *unit,
 
 }
 
+int config_parse_wireguard_private_key_file(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_free_ char *key = NULL;
+        Wireguard *w;
+        int r;
+
+        assert(data);
+        w = WIREGUARD(data);
+        assert(w);
+
+        r = read_one_line_file(rvalue, &key);
```

Comment:
hmm, read_one_line_file() is probably not the right function to use. I mean, base64 explicitly allows embedded line breaks which shall be ignored. Hence I figure we should use read_full_file() or so, and accept embedded newlines for this too